### PR TITLE
Doc/mkdocs creation

### PR DESCRIPTION
**Pull request by xNok**

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    - Premières Utiliations: userguide/firstUse.md
+    - Module gestion Associative: userguide/userManagement.md
+- Dev guide: devguide.md
+- Hacker guide:
+    - Git workflow: hackerguide/Git-workflow.md
+    - Dev VM: hackerguide/dev-env.md
+    - Configure your environment: hackerguide/Dev-tools.md
+    - Project coding guidelines: hackerguide/Coding-guidelines.md
+    - API Documentation: hackerguide/API-documentation.md
+- About: about.md
```
Comment (theofidry) [inline mkdocs.yml]: Add a return line here

File: docs/hackerguide/Git-workflow.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
+## Table of contents
+
+* [Fork Incipio](/hackerguide/Git-workflow#fork-incipio)
```
Comment (theofidry) [inline docs/hackerguide/Git-workflow.md]: Not sure with mkdocs, but for file navigation with GitHub, the links are relative to the file emplacement, i.e. here the link should be `#fork-incipio`.

**Comment by theofidry** [comment]
Far more clear :)

Is there a way to get a preview of the result so that we can see the issues encountered before merging?

**Comment by xNok** [comment]
First install mkdocs
>pip install mkdocs

Then run
>mkdocs serve

If you want the html vertion
>mkdocs build

**Comment by theofidry** [comment]
I've done a PR to fix a little bit the doc content.

What's left:

* [ ] Add travis script to push the doc on the branch `gh-pages`
* [ ] Add doc for building the mkdocs
* [ ] Ensure we can build the doc from a VM
* [ ] final review

**Comment by xNok** [comment]
Test : http://xnok.github.io/Incipio/

> Add travis script to push the doc on the branch gh-pages

Mkdocs has a command to deploy gh-page : 
`
mkdocs gh-deploy --clean
`
I'm not familiar with travis. Can we execute this commande (python library)? How can I test it locally ?


> Ensure we can build the doc from a VM

Y shouldn't use a VM to create the mkdocs, because it create a build-in web Server. Y'd rather run it on your machine.

**Comment by theofidry** [comment]
Yop @xNok.

> Mkdocs has a command to deploy gh-page : 
 mkdocs gh-deploy --clean 
I'm not familiar with travis. Can we execute this commande (python library)? How can I test it locally ?

A very simple way would be to enable Travis on your fork

>> Ensure we can build the doc from a VM

>Y shouldn't use a VM to create the mkdocs, because it create a build-in web Server. Y'd rather run it on your machine.

You're right, but it requires Python which may cause problem on some machines. I'm not saying we must do it this way, I'm saying that if someone has a vagrant machine, he should be able to compile the doc.

That being said it can be very annoying. If it's just too much of a pain in the ass leave those two points at that for now. I would rather finish the other points and doing the scenarios.

I was kind of very busy with my moving to London, I'll avance LoopBackApiBundle this week and come back to Incipio this weekend.

**Comment by theofidry** [comment]
@xNok I've integrated the doc :)

**Pull request closed without merging**